sklearn/metrics/tests/test_dist_metrics.py

```diff
@@ -73,6 +64,17 @@ def dist_func(x1, x2, p):
     )
 
 
+# TODO: remove this test in 1.3
+def test_neighbors_distance_metric_deprecation():
```

Comment:
This test already exist in `main`:

https://github.com/scikit-learn/scikit-learn/blob/1d1aadd0711b87d2a11c80aad15df6f8cf156712/sklearn/neighbors/tests/test_neighbors.py#L1841